src/Tools/babylon.observable.ts

```diff
@@ -273,7 +273,9 @@
             state.skipNextObservers = false;
             state.lastReturnValue = eventData;
 
-            for (var obs of this._observers) {
+            // Iterate over copy of array to handle removed observers during callback
+            var observerList = this._observers.slice();

```

Comment:
This cannot be kept. We cannot create an array on every call to observables. This will put too much pressure on GC as we extensively use observables
The pattern here is to either set observer.unregisterOnNextCall = true or call addOnce